solidity/contracts/ChainlinkLib.sol

```diff
@@ -0,0 +1,43 @@
+pragma solidity ^0.4.18;
+pragma experimental ABIEncoderV2;
+
+import "./Buffer.sol";
+import "./CBOR.sol";
+
+library ChainlinkLib {
+  using CBOR for Buffer.buffer;
+
+  struct Run {
+    bytes32 id;
+    bytes32 jobId;
+    address callbackAddress;
+    bytes4 callbackFunctionId;
+    Buffer.buffer buf;
+  }
+
+  function add(Run memory self, string _key, string _value)
+    internal
+  {
+    self.buf.encodeString(_key);
+    self.buf.encodeString(_value);
+  }
+
+  function addStringArray(Run memory self, string _key, string[] memory _values)
+    internal
+  {
+    self.buf.encodeString(_key);
+    self.buf.startArray();
+    for (uint256 i = 0; i < _values.length; i++) {
+      self.buf.encodeString(_values[i]);
+    }
+    self.buf.endSequence();
+  }
+
+  function close(Run memory self)
+    internal
+    returns (bytes)
+  {
+     self.buf.endSequence();
+     return self.buf.buf;
```

Comment:
It's worth noting that we `startMap` in the `Chainlinked` constructor, but explicitly close it here. Furthermore, we never use `id`, `jobId`, `callbackAddress/FunctionId` in this file and exclusively use `buf`. Perhaps this should follow the pattern used by `CBOR` and just add helper methods on top of `Buffer`, and we place the more CL domain specific functionality in `Chainlinked` alongside the ctor.